static/js/components/Reminders.jsx

```diff
@@ -0,0 +1,449 @@
+import MuiDialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import Close from "@mui/icons-material/Close";
+import Typography from "@mui/material/Typography";
+import AddIcon from "@mui/icons-material/Add";
+import grey from "@mui/material/colors/grey";
+
+// eslint-disable-next-line import/no-unresolved
+import Form from "@rjsf/material-ui/v5";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import { showNotification } from "baselayer/components/Notifications";
+
+import * as Actions from "../ducks/reminders";
+
+dayjs.extend(utc);
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    width: "100%",
+    overflow: "scroll",
+  },
+  eventTags: {
+    marginLeft: "0.5rem",
+    "& > div": {
+      margin: "0.25rem",
+      color: "white",
+      background: theme.palette.primary.main,
+    },
+  },
+  buttons: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  button: {
+    maxWidth: "1.2rem",
+  },
+  buttonIcon: {
+    maxWidth: "1.2rem",
+  },
+}));
+
+// Tweak responsive styling
+const getMuiTheme = (theme) =>
+  createTheme(
+    adaptV4Theme({
+      palette: theme.palette,
+      overrides: {
+        MUIDataTablePagination: {
+          toolbar: {
+            flexFlow: "row wrap",
+            justifyContent: "flex-end",
+            padding: "0.5rem 1rem 0",
+            [theme.breakpoints.up("sm")]: {
+              // Cancel out small screen styling and replace
+              padding: "0px",
+              paddingRight: "2px",
+              flexFlow: "row nowrap",
+            },
+          },
+          tableCellContainer: {
+            padding: "1rem",
+          },
+          selectRoot: {
+            marginRight: "0.5rem",
+            [theme.breakpoints.up("sm")]: {
+              marginLeft: "0",
+              marginRight: "2rem",
+            },
+          },
+        },
+      },
+    })
+  );
+
+const dialogTitleStyles = (theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  title: {
+    marginRight: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: grey[500],
+  },
+});
+
+const DialogTitle = withStyles(dialogTitleStyles)(
+  ({ children, classes, onClose }) => (
+    <MuiDialogTitle className={classes.root}>
+      <Typography variant="h6" className={classes.title}>
+        {children}
+      </Typography>
+      {onClose ? (
+        <IconButton
+          aria-label="close"
+          className={classes.closeButton}
+          onClick={onClose}
+        >
+          <Close />
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  )
+);
+
+const NewReminder = ({ resourceId, resourceType, handleClose }) => {
+  const dispatch = useDispatch();
+  const defaultDate = dayjs().utc().format("YYYY-MM-DDTHH:mm:ssZ");
```

Comment:
This default date will always be in the past by the time you finish filling the form. 
Maybe make the reminder in 1 day from now by default?

Reply:
Makes sense! I'll do that then.